File: DesignPatterns/01_GOF23/DP_examples/Python/03_Behavioral/Visitor/ConcreteVisitor.py
from Visitor import IComputerPartVisitor

# 具体访问者：ComputerPartDisplayVisitor
class ComputerPartDisplayVisitor(IComputerPartVisitor):
    def visit_computer(self, computer):
        print("Displaying Computer")
    
    def visit_mouse(self, mouse):
        print("Displaying Mouse")
        mouse.operation()
    
    def visit_keyboard(self, keyboard):
        print("Displaying Keyboard")
        keyboard.operation()
    
    def visit_monitor(self, monitor):
        print("Displaying Monitor")
        monitor.operation()

File: DesignPatterns/01_GOF23/DP_examples/Go/03_Behavioral/Visitor/concrete_visitor.go
package visitor

// ComputerPartDisplayVisitor 具体访问者：显示计算机部件
type ComputerPartDisplayVisitor struct{}

// NewComputerPartDisplayVisitor 创建显示访问者实例
func NewComputerPartDisplayVisitor() *ComputerPartDisplayVisitor {
	return &ComputerPartDisplayVisitor{}
}

// VisitComputer 访问计算机
func (v *ComputerPartDisplayVisitor) VisitComputer(computer *Computer) {
	println("Displaying Computer")
}

// VisitMouse 访问鼠标
func (v *ComputerPartDisplayVisitor) VisitMouse(mouse *Mouse) {
	println("Displaying Mouse")
	mouse.Operation()
}

// VisitKeyboard 访问键盘
func (v *ComputerPartDisplayVisitor) VisitKeyboard(keyboard *Keyboard) {
	println("Displaying Keyboard")
	keyboard.Operation()
}

// VisitMonitor 访问显示器
func (v *ComputerPartDisplayVisitor) VisitMonitor(monitor *Monitor) {
	println("Displaying Monitor")
	monitor.Operation()
}
